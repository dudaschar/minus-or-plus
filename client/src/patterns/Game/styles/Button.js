import styled from  'styled-components';

import Button from 'src/components/Button/Button';

export const StyledButton = styled(Button)`
  margin: ${({ theme }) => `0 ${theme.spacing.small}`};
`;

StyledButton.displayName = 'StyledButton';
