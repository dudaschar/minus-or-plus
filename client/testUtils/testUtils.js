import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../stylesUtils/theme';

const TestingWithProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

TestingWithProviders.propTypes = {
  children: PropTypes.any.isRequired,
};

const customRender = (ui, options) => (
  render(ui,  { wrapper: TestingWithProviders, ...options })
);

export * from '@testing-library/react';
export { customRender as render };

