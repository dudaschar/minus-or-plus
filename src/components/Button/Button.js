import React from 'react';
import PropTypes from 'prop-types';

import { BaseButton } from './styles/BaseButton';

function Button(props) {
  return (
    <BaseButton
      {...props}
    />
  );
}

Button.defaultProps = {
  disabled: false,
};

Button.propTypes = {
  disabled: PropTypes.bool,
};

export default Button;