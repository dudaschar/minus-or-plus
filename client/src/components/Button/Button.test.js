import React from 'react';
import { render } from 'testUtils/testUtils';

import Button from './Button';

describe('Button Component', () => {
  it('should render the component properly', () => {
    const { asFragment } = render(
      <Button>Click</Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});