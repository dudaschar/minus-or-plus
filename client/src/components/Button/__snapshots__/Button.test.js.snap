// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render the component properly 1`] = `
<DocumentFragment>
  <button
    class="BaseButton-vk7j35-0 jhOaS"
  >
    Click
  </button>
</DocumentFragment>
`;
